src/main/java/main/movie/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package main.movie.service.impl;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import main.movie.dao.OrderDao;
+import main.movie.lib.Inject;
+import main.movie.lib.Service;
+import main.movie.model.Order;
+import main.movie.model.ShoppingCart;
+import main.movie.model.Ticket;
+import main.movie.model.User;
+import main.movie.service.OrderService;
+import main.movie.service.ShoppingCartService;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+
+    @Inject
+    private OrderDao orderDao;
+    @Inject
+    private ShoppingCartService shoppingCartService;
+
+    @Override
+    public Order completeOrder(User user) {
+        ShoppingCart cart = shoppingCartService.getByUser(user);
+        List<Ticket> tickets = cart.getTickets();
```

Comment:
Does it work without deep cloning?
Since it's a reference 🤔 Curious

Reply:
I have doubts too. Send a screenshot of hibernate sql queries while orderDao.add(order) method